L1Trigger/L1TCaloLayer1/src/UCTLayer1.cc

```diff
@@ -0,0 +1,117 @@
+#include <iostream>
+#include <iomanip>
+#include <stdlib.h>
+#include <stdint.h>
+
+#include "UCTLayer1.hh"
+
+#include "UCTCrate.hh"
+#include "UCTCard.hh"
+#include "UCTRegion.hh"
+#include "UCTTower.hh"
+
+#include "UCTGeometry.hh"
+
+UCTLayer1::UCTLayer1() : uctSummary(0) {
+  UCTGeometry g;
+  for(uint32_t crate = 0; crate < g.getNCrates(); crate++) {
+    crates.push_back(new UCTCrate(crate));
+  }
+}
+
+UCTLayer1::~UCTLayer1() {
+  for(uint32_t i = 0; i < crates.size(); i++) {
+    if(crates[i] != 0) delete crates[i];
+  }
+}
+
+bool UCTLayer1::clearEvent() {
+  for(uint32_t i = 0; i < crates.size(); i++) {
+    if(crates[i] != 0) crates[i]->clearEvent();
+  }
+  return true;
+}
+
+const UCTRegion* UCTLayer1::getRegion(int regionEtaIndex, uint32_t regionPhiIndex) const {
+  if(regionEtaIndex == 0 || regionEtaIndex < -7 || regionEtaIndex > 7 ||
+     regionPhiIndex <= 0 || regionPhiIndex >= 19) {
+    return 0;
+  }
+  // Get (0,0) tower region information
+  UCTGeometry g;
+  UCTRegionIndex r = UCTRegionIndex(regionEtaIndex, regionPhiIndex);
+  UCTTowerIndex t = g.getUCTTowerIndex(r);
+  uint32_t absCaloEta = abs(t.first);
+  uint32_t absCaloPhi = abs(t.second);
+  uint32_t crt = g.getCrate(absCaloEta, absCaloPhi);
+  if(crt >= crates.size()) {
+    std::cerr << "UCTLayer1::getRegion - Crate number is wrong - " << std::hex << crt 
```

Comment:
> On Mar 22, 2016, at 1:58 PM, Sridhara Dasu <notifications@github.com> wrote:
> 
> In L1Trigger/L1TCaloLayer1/src/UCTLayer1.cc:
> 
> > +}
> > +
> > +const UCTRegion* UCTLayer1::getRegion(int regionEtaIndex, uint32_t regionPhiIndex) const {
> > +  if(regionEtaIndex == 0 || regionEtaIndex < -7 || regionEtaIndex > 7 ||
> > +     regionPhiIndex <= 0 || regionPhiIndex >= 19) {
> > +    return 0;
> > +  }
> > +  // Get (0,0) tower region information
> > +  UCTGeometry g;
> > +  UCTRegionIndex r = UCTRegionIndex(regionEtaIndex, regionPhiIndex);
> > +  UCTTowerIndex t = g.getUCTTowerIndex(r);
> > +  uint32_t absCaloEta = abs(t.first);
> > +  uint32_t absCaloPhi = abs(t.second);
> > +  uint32_t crt = g.getCrate(absCaloEta, absCaloPhi);
> > +  if(crt >= crates.size()) {
> > +    std::cerr << "UCTLayer1::getRegion - Crate number is wrong - " << std::hex << crt 
> 
> Rant:
> 
> I have an issue with this message logger policing. What this message logger requirement does is that, it makes our code a pain in the butt to use in non-CMSSW environment. This code is visualized for online use as well, and we may want to run it even in peta-linux environment on our ZYNQ based boards. Why does CMS ban std::cerr usage? I understand wanting std::out putting crap in every log file, but rare error reporting can be excused when useful for specific projects in my opinion.
> 

the typical solution is ifdefs - the syntax is the same between cerr and message logger.


> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>